# Exploring Gun Deaths in The US

In [4]:
# Importing and taking a peek into that data
import csv

with open('guns.csv', 'r') as f:
    file = csv.reader(f)
    data = list(file)
    
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


# Removing Headers from the Data List

In [5]:
# Remove the header from the list leaving the data
header = data[0]
data.remove(header)
print(data[:5])

[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


# Gun Deaths By Year

In [3]:
# Count how many deaths occur in each year
year_counts = {}
years = [i[1] for i in data]

for i in years:
    if i not in year_counts:
        year_counts[i] = 1
    else:
        year_counts[i] += 1
year_counts

{'2012': 33563, '2013': 33636, '2014': 33599}

# Gun Deaths by Month and Year 

In [4]:
import datetime
dates = [datetime.datetime(year = int(i[1]), month = int(i[2]), day = 1) for i in data]
dates[0:5]

[datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 1, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0),
 datetime.datetime(2012, 2, 1, 0, 0)]

In [5]:
# Count how many deaths occur in each month
date_counts = {}
for i in dates:
    if i not in date_counts:
        date_counts[i] = 1
    else:
        date_counts[i] += 1

date_counts

{datetime.datetime(2012, 1, 1, 0, 0): 2758,
 datetime.datetime(2012, 2, 1, 0, 0): 2357,
 datetime.datetime(2012, 3, 1, 0, 0): 2743,
 datetime.datetime(2012, 4, 1, 0, 0): 2795,
 datetime.datetime(2012, 5, 1, 0, 0): 2999,
 datetime.datetime(2012, 6, 1, 0, 0): 2826,
 datetime.datetime(2012, 7, 1, 0, 0): 3026,
 datetime.datetime(2012, 8, 1, 0, 0): 2954,
 datetime.datetime(2012, 9, 1, 0, 0): 2852,
 datetime.datetime(2012, 10, 1, 0, 0): 2733,
 datetime.datetime(2012, 11, 1, 0, 0): 2729,
 datetime.datetime(2012, 12, 1, 0, 0): 2791,
 datetime.datetime(2013, 1, 1, 0, 0): 2864,
 datetime.datetime(2013, 2, 1, 0, 0): 2375,
 datetime.datetime(2013, 3, 1, 0, 0): 2862,
 datetime.datetime(2013, 4, 1, 0, 0): 2798,
 datetime.datetime(2013, 5, 1, 0, 0): 2806,
 datetime.datetime(2013, 6, 1, 0, 0): 2920,
 datetime.datetime(2013, 7, 1, 0, 0): 3079,
 datetime.datetime(2013, 8, 1, 0, 0): 2859,
 datetime.datetime(2013, 9, 1, 0, 0): 2742,
 datetime.datetime(2013, 10, 1, 0, 0): 2808,
 datetime.datetime(2013, 11,

# Gun Death by Race and Sex

In [6]:
# Count the number of male and female deaths
sex = [i[5] for i in data]
sex_counts = {}
for i in sex:
    if i not in sex_counts:
        sex_counts[i] = 1
    else:
        sex_counts[i] += 1

sex_counts

{'F': 14449, 'M': 86349}

In [7]:
# Count the number of deaths by race
race = [i[7] for i in data]
race_counts = {}
for i in race:
    if i not in race_counts:
        race_counts[i] = 1
    else:
        race_counts[i] += 1

race_counts

{'Asian/Pacific Islander': 1326,
 'Black': 23296,
 'Hispanic': 9022,
 'Native American/Native Alaskan': 917,
 'White': 66237}

In [8]:
# Count the proportion of deaths by race
total_deaths = 0
for key, val in race_counts.items():
    total_deaths += val
    
total_gender = 0
for key, val in sex_counts.items():
    total_gender += val


In [9]:
race_ratio = {}
for key, val in race_counts.items():
    if key not in race_ratio:
        race_ratio[key] = val / total_deaths
        
race_ratio

{'Asian/Pacific Islander': 0.01315502291712137,
 'Black': 0.23111569674001467,
 'Hispanic': 0.0895057441615905,
 'Native American/Native Alaskan': 0.009097402726244569,
 'White': 0.6571261334550289}

In [10]:
# Count the proportion of deaths by Sex
sex_ratio = {}
for key, val in sex_counts.items():
    if key not in sex_ratio:
        sex_ratio[key] = val / total_gender
sex_ratio

{'F': 0.1433460981368678, 'M': 0.8566539018631322}

# Findings

65.7% of people killed by guns are white,  23.1% are 'Black', 8.9% are 'Hispanic', 1.3% are 'Asian/Pacific Islander while 0.9% are "Native American".

Only people categorised as 'Black' have a  higher death ratio relative to their proportion of the US poplation as estimated for 2015 (ref: https://www.census.gov/quickfacts/). Blacks represent 13.3% of the population while 23.1% of gun related deaths are people categorised as Black. What are the major causes of death among 'Black' people? It will be interesting to see if police related deaths are signicant and disproportionate amongst Blacks as it is widely believed by many black people.

Also, Males have much higher likelihood of being killed by guns 85.6% compared to 14.3 percent for females even though females represent 50.8% of the population in 2015.

# Introducing the Census Data

In [11]:
# Import the Census Data
f = open("census.csv", 'r')
file = csv.reader(f)
census = list(file)
print(census)

[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


# Rates of Gun Deaths by Race

In [12]:
mapping = {}
for key, val in race_counts.items():
    if key == 'Asian/Pacific Islander':
        mapping[key] = int(census[1][14]) + int(census[1][15])
    elif key =='Black':
        mapping[key] = int(census[1][12])
    elif key == 'Hispanic':
        mapping[key] = int(census[1][11])
    elif key == 'Native American/Native Alaskan':
        mapping[key] = int(census[1][13])
    elif key == 'White':
        mapping[key] = int(census[1][10])
        
mapping

        
    

        
    

{'Asian/Pacific Islander': 15834141,
 'Black': 40250635,
 'Hispanic': 44618105,
 'Native American/Native Alaskan': 3739506,
 'White': 197318956}

In [13]:
race_per_hundredk = {}
for key in race_counts:
    race_per_hundredk[key] = (race_counts[key] / mapping[key] ) * 100000         

In [14]:
race_per_hundredk

{'Asian/Pacific Islander': 8.374309664161762,
 'Black': 57.8773477735196,
 'Hispanic': 20.220491210910907,
 'Native American/Native Alaskan': 24.521955573811088,
 'White': 33.56849303419181}

# Filter by 'Intent' Column 

In [15]:
intents = [i[3] for i in data]
races = [i[7] for i in data]
homicide_race_per_hundredk = {}
for i, race in enumerate(races):
    if intents[i] == 'Homicide':
        if race not in homicide_race_per_hundredk:
            homicide_race_per_hundredk[race] = 1
        else:
            homicide_race_per_hundredk[race] += 1

In [16]:
homicide_race_per_hundredk

{'Asian/Pacific Islander': 559,
 'Black': 19510,
 'Hispanic': 5634,
 'Native American/Native Alaskan': 326,
 'White': 9147}

In [17]:
for i in homicide_race_per_hundredk:
    homicide_race_per_hundredk[i] = (homicide_race_per_hundredk[i] / mapping[i] ) * 100000

In [18]:
homicide_race_per_hundredk

{'Asian/Pacific Islander': 3.530346230970155,
 'Black': 48.471284987180944,
 'Hispanic': 12.627161104219914,
 'Native American/Native Alaskan': 8.717729026240365,
 'White': 4.6356417981453335}

# Findings

3 people of Asian/Pacific island origin per 100,000 lose theeir life through gun related homicide while 48 black people per 100,000 of black people lose their life to gun related homicide. The rate per 100,000 of people from similar racial category is 12 9 and 4 for Hispanic, Native American and Whites respectively

The proportion of Gun relate

We shall explore our data once more to see the rate of lives lost to police guns

# Death by Police Guns

In [19]:
police_race_deaths = {}
police_guns = [i[4] for i in data]
for i, val in enumerate(races):
    if police_guns[i] == '1':
        if val not in police_race_deaths:
            police_race_deaths[val] = 1
        else:
            police_race_deaths[val] += 1
            
            
    
police_race_deaths

{'Asian/Pacific Islander': 30,
 'Black': 356,
 'Hispanic': 282,
 'Native American/Native Alaskan': 25,
 'White': 709}

### Death by Police per 100,000 of Population per Race 

In [20]:
#calculate number of deaths caused by police per 100,000 of racial population
police_race_death_per_hundredk = {}
for i in police_race_deaths:
    police_race_death_per_hundredk[i] = (police_race_deaths[i] / mapping[i]) * 100000

police_race_death_per_hundredk

{'Asian/Pacific Islander': 0.18946401955117112,
 'Black': 0.8844580961269306,
 'Hispanic': 0.6320304280067475,
 'Native American/Native Alaskan': 0.6685375020122979,
 'White': 0.3593167196769478}

# Findings 

Approximately 1 in 100,000 (0.88 to be precise) black people was killed by police shooting, this is the highest amongst all the races while white people had the lowest death per 100,000 people (0.35 to be precise)